Code/Framework/AzCore/Tests/Settings/SettingsRegistryTests.cpp

```diff
@@ -1473,16 +1439,125 @@ namespace SettingsRegistryTests
         auto outcome = m_registry->MergeSettingsFile(path.Native(), AZ::SettingsRegistryInterface::Format::JsonMergePatch, {});
         EXPECT_TRUE(outcome);
 
-        AZStd::string history;
-        bool result = m_registry->Get(history, AZ_SETTINGS_REGISTRY_HISTORY_KEY "/0");
-        EXPECT_TRUE(result);
-        EXPECT_STREQ(path.c_str(), history.c_str());
-
         AZ::s64 intValue{};
         EXPECT_TRUE(m_registry->Get(intValue, "/TestObject/IntValue"));
         EXPECT_EQ(7, intValue);
     }
 
+    TEST_F(SettingsRegistryTest, MergeSettingsFile_JsonWithImport_MergesImportedFiles_InOrder)
+    {
+        auto filePath = CreateTestFile("Memory.iphone13.setreg", R"({ "pre_field": {"first": 7 },
+            "$import": "Memory.ios.setreg", "non_override": 7, "post_field": {"2nd": "twenty-one"} })");
+        CreateTestFile("Memory.ios.setreg", R"({ "$import": { "filename": "Memory.mobile.setreg" },
+            "non_override2": "Hello World", "override_imported_settings": "override" })");
```

Comment:
Is it ok to use the names "ios" and "iphone" on all platforms like this? I'd suggest using more generic names in the test (ex: "os", "phone") just in case.

Reply:
Well these filesnames won't affect any functionality at all and they aren't being loaded as part of a `SettingsRegistryInterface::MergeSettingsFolder` call.
I'll generalized the names anyway.

Reply:
I wasn't concerned about functionality, just that it's generally better to avoid trademarked names.